game.py

```diff
@@ -0,0 +1,302 @@
+from math import inf
+from copy import copy, deepcopy
+import random
+
+class GameState():
+    def __init__(self, board, player_next, previous_game_state = None):
+        self._board = board
+        self._player_next = player_next
+        self._previous_game_state = previous_game_state
+
+    def _copy(self):
+        board = deepcopy(self._board)
+        player_next = self._player_next
+        previous_game_state = deepcopy(self)
+        return GameState(board, player_next, previous_game_state)
+
+                
+    def get_board(self):
+        return self._board
+    
+    def get_player_next(self):
+        return self._player_next
+
+    def make_move(self, move):
+        new_game_state = self._copy()
+        new_game_state._board[move[0]][move[1]] = new_game_state._player_next
+        new_game_state._player_next = - new_game_state._player_next
+        return new_game_state
+        
+    def unmake_move(self):
+        new_game_state = self._previous_game_state
+        return new_game_state
+        
+    def get_next_moves(self, moves_sort = False, game_end = False):
+        moves = []
+        any_moves = False
+        for i in range(len(self._board)):
+            if any_moves:
+                break
+            for j in range(len(self._board[i])):
+                if self._board[i][j] == None:
+        #            moves.append ((i, j))
+                    moves.insert(0, (i, j))
+                    if game_end:
+                        any_moves = True
+                        break
+                    
+        random.shuffle(moves)
+        
+        if moves_sort:
+            moves_and_scores = []
+            i = 0
+            for move in moves:

```

Comment:
Take a look at `enumerate()`, useful when you want an index as well (like here), so you can just do `for i, move in enumerate(moves): ...`